### PR TITLE
feat(output): add report output format

### PR DESCRIPTION
**Pull request by dclayton-godaddy**
To help us get this pull request reviewed and merged quickly, please be sure to include the following items:

* [x] Tests (if applicable)
* [ ] Documentation (if applicable)
* [ ] Changelog entry
* [x] A full explanation here in the PR description of the work done

## PR Type
What kind of change does this PR introduce?

* [ ] Bugfix
* [x] Feature
* [ ] Code style update (formatting, local variables)
* [ ] Refactoring (no functional changes, no api changes)
* [x] Build related changes
* [x] CI related changes
* [ ] Documentation content changes
* [ ] Tests
* [ ] Other

## Backward Compatibility

Is this change backward compatible with the most recently released version? Does it introduce changes which might change the user experience in any way? Does it alter the API in any way?

* [ ] Yes (backward compatible)
* [x] No (breaking changes)

## What's new?
- added report format to output a reviewer-friendly report
- drop support for python 3.6
- add support for pyenv in tox
- fix Docker build

```text
$ tartufo --output-format report scan-local-repo .
Tartufo Scan Results (Time: 2022-11-09T16:24:34.012693)
All clear. No secrets detected.

Configuration:
  version:             3.2.1
  entropy:             Enabled
    sensitivity: 75
  regex:               Enabled

Excluded paths:
  package-lock.json: Autogenerated package lock file
  cicd.yaml: cert thumbprint
  README.md: readme example

Excluded signatures:
  a32cedf592e11c8d2efd06e4bd7906b340b64cb2f4b1cbb925fe908042d5d7bb: simple test file

Excluded entropy patterns:
  thumbprint:\s{1}[a-zA-Z0-9]{40} (path=cicd.yaml, scope=line, type=search): cert thumbprint
```

### CODE REVIEW

**Comment by mgaspar-godaddy** [comment]
I like it. This would help greatly when someone needs to submit their scan results to someone else to be able to quickly and easily know what each exclusions is for and greatly speed up any review process in place.

**Review by sjacoby-godaddy — APPROVED** [review]
looks good

File: pyproject.toml [hunk 1]
```diff
@@ -39,11 +39,12 @@ tartufo = "tartufo.cli:main"
...
 GitPython = "<3.1.20"
-pygit2 = [ {version = "~1.6", python = "~3.6"}, {version = "^1.9.0", python = "^3.7"} ]
-click = [{version = "~8.0.4", python = "~3.6"}, {version = "^8.1.0", python = "^3.7"}]
+pygit2 = [{version = "^1.9.0", python = "^3.7"}]
+click = [{version = "^8.1.0", python = "^3.7"}]
 colorama = {version = "*", markers = "sys_platform == 'win32'"}
 dataclasses = {version = "*", python = "< 3.7"}
-python = "^3.6.2"
+importlib-metadata = {version = "4.8.3", python = "< 3.8"}
+python = "^3.7"
```
Comment (sushantmimani) [inline pyproject.toml:47]: It'll be preferable if we keep the "drop support for 3.6" changes out of this PR, unless there's something that necessitates this change in this PR.

**Review by sushantmimani — CHANGES_REQUESTED** [review]
"Drop support for 3.6" should be a separate PR.

File: tartufo/cli.py [hunk 2]
```diff
@@ -122,6 +122,7 @@ def get_command(self, ctx: click.Context, cmd_name: str) -> Optional[click.Comma
             types.OutputFormat.Json.value,
             types.OutputFormat.Compact.value,
             types.OutputFormat.Text.value,
+            types.OutputFormat.Report.value,
```
Comment (schoi-godaddy) [inline tartufo/cli.py:125]: Could we add this to below help text as well please?

**Review by schoi-godaddy — COMMENTED** [review]
Like the new format, ty.

**Pull request closed without merging**